### PR TITLE
Python: Update `XADD` and `XTRIM` Options

### PR DESCRIPTION
**Pull request by acarbonetto**

### CODE REVIEW

File: python/python/glide/async_commands/core.py [hunk 1]
```diff
@@ -205,37 +209,71 @@ class TrimByMinId(StreamTrimOptions):
...
-        super().__init__(exact, threshold, "MINID", limit)
+        super().__init__(threshold, "MINID", exact, None)
+
+    def __init__(self, threshold: str, limit: int):
+        """
+        Initialize trim option by minimum ID.
+
+        Args:
+            threshold (str): Threshold for trimming by minimum ID.
+            limit (int): Max number of entries to be trimmed. Defaults to None.
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:238]: There is another constructor without limit, maybe skip that sentence? Also, limit is not optional, so user can't pass None.
```suggestion
            limit (int): Max number of entries to be trimmed.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 * Python: Added ZDIFFSTORE command ([#1378](https://github.com/aws/glide-for-redis/pull/1378))
 * Python: Added ZDIFF command ([#1401](https://github.com/aws/glide-for-redis/pull/1401))
 * Python: Added BZPOPMIN and BZPOPMAX commands ([#1399](https://github.com/aws/glide-for-redis/pull/1399))
+* Python: Update XADD and XTRIM options (TODO)
```
Comment (Yury-Fridlyand) [inline CHANGELOG.md:29]: I think it should be in breacking changes section below

**Review by Yury-Fridlyand — COMMENTED** [review]
CI is red

File: CHANGELOG.md (hunk 2 above)
Comment (acarbonetto) [reply CHANGELOG.md:29]: Will add this PR to line 23, which is new for the release.

File: python/python/tests/test_async_client.py [hunk 3]
```diff
@@ -2636,26 +2636,51 @@ async def test_xadd_xtrim(self, redis_client: TRedisClient):
...
+            assert (
+                await redis_client.xadd(
+                    key,
+                    [(field + counter, value + counter)],
+                    StreamAddOptions(id=id),
+                )
+                == id
+            )
+
+        assert await redis_client.custom_command(["XLEN", key]) == 6
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_async_client.py:2676]: nit: leave TODOs for `xlen` for future us

**Review by Yury-Fridlyand — COMMENTED** [review]
CI is still red

**Review by Yury-Fridlyand — APPROVED** [review]

File: python/python/glide/async_commands/core.py [hunk 4]
```diff
@@ -205,37 +209,96 @@ class TrimByMinId(StreamTrimOptions):
...
             threshold (str): Threshold for trimming by minimum ID.
-            limit (Optional[int]): Max number of entries to be trimmed. Defaults to None.
-                Note: If `exact` is set to `True`, `limit` cannot be specified.
         """
-        super().__init__(exact, threshold, "MINID", limit)
+        print(threshold)
+        return cls(threshold, "MINID", exact, None)
+
+    @classmethod
+    def create_withLimit(cls, threshold: str, limit: int):
```
Comment (jonathanl-bq) [inline python/python/glide/async_commands/core.py:245]: Is there a precedent for using camelCase midway through a function name in Python? I thought every method was supposed to be snake_case. I'm okay with it being this way if we're consistent about it, but just want to be sure.

**Review by jonathanl-bq — APPROVED** [review]
One minor nit, but otherwise seems fine.

File: python/python/glide/async_commands/core.py (hunk 4 above)
Comment (acarbonetto) [reply python/python/glide/async_commands/core.py:245]: It odd.  They use snake_case but are inconsistent with underscores, like `zrange_withscores`.  
I will update this to `create_withlimit` and see what Shoham things.

**Pull request merged**